plugins/org.genivi.faracon.cli/src/org/genivi/faracon/cli/CommandLineHandler.java

```diff
@@ -24,104 +25,71 @@ public CommandLineHandler()
-			// -ns --no-stub do not generate stub code
-			if (parsedArguments.hasOption("ns")) {
-				cliTool.setNoStubCode();
-			}
-
-			// destination: -d --dest overwrite default directory
-			if (parsedArguments.hasOption("d")) {
-				cliTool.setDefaultDirectory(parsedArguments.getOptionValue("d"));
-			}
-
-			// destination: -dsub --dest-subdirs use destination/<filename>/ subdirs
-			// as dest directory
-			if(parsedArguments.hasOption("dsub")) {
-				cliTool.setDestinationSubdirs();
-			}
-
-			// destination: -dc --dest-common overwrite target directory for
-			// common part
-			if (parsedArguments.hasOption("dc")) {
-				cliTool.setCommonDirectory(parsedArguments.getOptionValue("dc"));
-			}
-
-			// destination: -dp --dest-proxy overwrite target directory for
-			// proxy code
-			if (parsedArguments.hasOption("dp")) {
-				cliTool.setProxyDirectory(parsedArguments.getOptionValue("dp"));
-			}
-
-			// destination: -ds --dest-stub overwrite target directory for stub
-			// code
-			if (parsedArguments.hasOption("ds")) {
-				cliTool.setStubDirectory(parsedArguments.getOptionValue("ds"));
-			}
-
-			// A file path, that points to a file, that contains the license
-			// text.
-			// -l --license license text in generated files
-			if (parsedArguments.hasOption("l")) {
-				cliTool.setLicenseText(parsedArguments.getOptionValue("l"));
-			}
-
-			// Switch off validation
-			if (parsedArguments.hasOption("nv")) {
-				cliTool.disableValidation();
-			}
-			// Switch off code generation at all
-			if (parsedArguments.hasOption("ng")) {
-				cliTool.disableCodeGeneration();
-			}
-			// Don't generate code for included types and interfaces
-			if (parsedArguments.hasOption("wod")) {
-				cliTool.noCodeforDependencies();
-			}
-			// Don't generate synchronous calls
-			if (parsedArguments.hasOption("nsc")) {
-				cliTool.disableSyncCalls();
-			}
-			// print out generated files
-			if (parsedArguments.hasOption("pf")) {
-				cliTool.listGeneratedFiles();
-			}
-			// finally invoke the generator.
-			return cliTool.generateDBus(files);
-		} else {
-			System.out.println("A *.fidl or *.fdepl file was not specified !");
+
+		List<String> commandLineArguments = parsedArguments.getArgList();
+		for (String commandLineArgument : commandLineArguments) {
+			ConsoleLogger.logWarning("Unexpected command line argument \"" + commandLineArgument + "\"");			
 		}
+
+		String[] francaFilePaths = parsedArguments.getOptionValues('f');
+		String[] araFilePaths = parsedArguments.getOptionValues('a');
+//		// We expect at least one fidl/fdepl file as command line argument
+//		if (files.size() > 0 && files.get(0) != null) {
+		if ((francaFilePaths == null || francaFilePaths.length == 0) && (araFilePaths == null || araFilePaths.length == 0)) {
+			ConsoleLogger.logError("At least one input model file has to be given!");			
+		}		
+		if ((francaFilePaths != null && francaFilePaths.length > 0) && (araFilePaths != null && araFilePaths.length > 0)) {
```

Comment:
Introducing two boolean local vars "haveFrancaInput" and "haveARAInput" would remove redundancy from lines 59 and 62.